flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/UnsignedTypeConversionITCase.java

```diff
@@ -43,136 +44,81 @@
 
     private static final Logger logger =
             LoggerFactory.getLogger(UnsignedTypeConversionITCase.class);
+    private static final DockerImageName MYSQL_57_IMAGE = DockerImageName.parse("mysql:5.7.34");
     private static final String DEFAULT_DB_NAME = "test";
     private static final String TABLE_NAME = "unsigned_test";
-    private static final int INITIALIZE_DB_MAX_RETRY = 3;
-    private static DB db;
-    private static String dbUrl;
-    private static Connection connection;
 
+    private static final String user = "root";
+    private static final String password = "";
+
+    private static final Map<String, String> DEFAULT_CONTAINER_ENV_MAP =
+            new HashMap<String, String>() {
+                {
+                    put("MYSQL_ROOT_HOST", "%");
+                }
+            };
+
+    public static final MySQLContainer<?> MYSQL_CONTAINER =
+            new MySQLContainer<>(MYSQL_57_IMAGE)
+                    .withEnv(DEFAULT_CONTAINER_ENV_MAP)
+                    .withUsername(user)
+                    .withPassword(password)
+                    .withLogConsumer(new Slf4jLogConsumer(logger));
+
+    private static String baseUrl;
+    private static Connection connection;
     private StreamTableEnvironment tEnv;
 
     @BeforeClass
-    public static void prepareMariaDB() throws IllegalStateException {
-        boolean initDbSuccess = false;
-        int i = 0;
-        // The initialization of maria db instance is a little unstable according to past CI tests.
-        // Add retry logic here to avoid initialization failure.
-        while (i < INITIALIZE_DB_MAX_RETRY) {
-            try {
-                db = DB.newEmbeddedDB(DBConfigurationBuilder.newBuilder().build());
-                db.start();
-                dbUrl = db.getConfiguration().getURL(DEFAULT_DB_NAME);
-                connection = DriverManager.getConnection(dbUrl);
-                try (Statement statement = connection.createStatement()) {
-                    statement.execute("CREATE DATABASE IF NOT EXISTS `" + DEFAULT_DB_NAME + "`;");
-                    ResultSet resultSet =
-                            statement.executeQuery(
-                                    "SELECT SCHEMA_NAME FROM "
-                                            + "INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '"
-                                            + DEFAULT_DB_NAME
-                                            + "';");
-                    if (resultSet.next()) {
-                        String dbName = resultSet.getString(1);
-                        initDbSuccess = DEFAULT_DB_NAME.equalsIgnoreCase(dbName);
-                    }
-                }
-            } catch (Exception e) {
-                logger.warn("Initialize DB failed.", e);
-                stopDb();
-            }
-            if (initDbSuccess) {
-                break;
+    public static void launchContainer() throws Exception {
+        MYSQL_CONTAINER.start();
```

Comment:
Rather than manually starting the container, can you just use JUnit `@Rule` like shown in this doc? https://github.com/testcontainers/testcontainers-java/blob/master/modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java#L42

This ties to the junit lifecycle, fixing for you all the issues related to start/stop leaks